src/ObjCRuntime/Runtime.cs

```diff
@@ -575,9 +575,6 @@ static unsafe IntPtr GetMethodFromToken (uint token_ref)
 
 		static unsafe IntPtr GetGenericMethodFromToken (IntPtr obj, uint token_ref)
 		{
-#if MONOMAC
-			throw new NotSupportedException ();
```

Comment:
XI/Classic didn't support generics either. If generics was the issue, the ifdef would have been `#if !XAMCORE_2_0`, not `#if MONOMAC`.

Also I can't see how removing an unconditional exception can break things... and the static registrar will show errors at compile time for generics in Classic mode.

My guess is that it was a temporary ifdef I added while fixing something to make the code compile, and then the ifdef was never removed.